### PR TITLE
Add screen name to preconditions

### PR DESCRIPTION
**Pull request by narenkmanoharan**
Add screen name to preconditions when an exception is thrown for when the screen is not present in backstack

### CODE REVIEW

File: magellan-library/src/test/java/com/wealthfront/magellan/ScreenTest.java [hunk 1]
```diff
@@ -149,20 +148,4 @@ public void whenTransitionFinished_afterTransitionFinished() {
     screen.whenTransitionFinished(listener);
     verify(listener).onTransitionFinished();
   }
-
-  private static class DummyScreen extends Screen<BaseScreenView<DummyScreen>> {
```
Comment (cmathew) [inline magellan-library/src/test/java/com/wealthfront/magellan/ScreenTest.java]: nice pulling this out for re-use. Minor: looks like the same code exists in ScreenGroupTest.java if you're willing to pull it out of there as well?

File: magellan-library/src/main/java/com/wealthfront/magellan/Navigator.java [hunk 2]
```diff
@@ -459,7 +459,7 @@ public void goBackTo(final Screen screen, NavigationType navigationType) {
     navigate(new HistoryRewriter() {
       @Override
       public void rewriteHistory(Deque<Screen> history) {
-        checkArgument(history.contains(screen), "Can't go back to a screen that isn't in history.");
+        checkArgument(history.contains(screen), String.format("Can't go back to a screen (%s) that isn't in history.", screen.getClass().getName()));
```
Comment (cmathew) [inline magellan-library/src/main/java/com/wealthfront/magellan/Navigator.java]: do we want to include the current history as well?

File: magellan-library/src/test/java/com/wealthfront/magellan/ScreenTest.java (hunk 1 above)
Comment (narenkmanoharan) [reply magellan-library/src/test/java/com/wealthfront/magellan/ScreenTest.java]: d

File: magellan-library/src/main/java/com/wealthfront/magellan/Navigator.java (hunk 2 above)
Comment (narenkmanoharan) [reply magellan-library/src/main/java/com/wealthfront/magellan/Navigator.java]: d

**Pull request merged**